xonsh/ptk2/shell.py

```diff
@@ -146,6 +147,14 @@ def singleline(
 
             prompt_args["style"] = style
 
+            style_overrides_env = env.get("PTK_STYLE_OVERRIDES")
+            if style_overrides_env:
+                try:
```

Comment:
What is the reason for this broad try/catch?

Reply:
1) The PTK_STYLE_OVERRIDES can be pretty much anything (it's a user input).
2) I didn't really take the time to check what can be thrown from `Style.from_dict` and `merge_styles`.

Reply:
True, but I think it better to let it fail than to suppress all errors. 

If you want to verify the content of the `PTK_STYLE_OVERRIDES` you should add an entry to the [`DEFAULT_ENSURERS`](https://github.com/xonsh/xonsh/blob/c97fc91c284f4a2b7fd57c7d1508afe71373e45c/xonsh/environ.py#L164-L273) 

These functions are used to validate the content when a value is set: 
See for example the [__setitem__](https://github.com/xonsh/xonsh/blob/c97fc91c284f4a2b7fd57c7d1508afe71373e45c/xonsh/environ.py#L1157-L1174)

Reply:
> True, but I think it better to let it fail than to suppress all errors.

The eventual exception get printed in the handler.

> If you want to verify the content of the PTK_STYLE_OVERRIDES you should add an entry to the DEFAULT_ENSURERS

I can do it, but I don't really think it makes much sense, because it's a dictionary of string keys with string values. Both are interpreted at some point somehow and both can contain possibly invalid values.

Reply:
Well, it makes sense if you want these settings to pass to a xonsh shell that is being run from another xonsh shell.